pages/calendar.js

```diff
@@ -1,39 +1,73 @@
 "use client"
-import React from 'react';
+import React, { useState } from 'react';
 import 'react-calendar/dist/Calendar.css';
-import activitiesData from '@/lib/db';
+import useSWR from 'swr';
 import dynamic from 'next/dynamic';
 import styled from 'styled-components';
 
+const fetcher = (url) => fetch(url).then((res) => res.json());
 const Calendar = dynamic(() => import('react-calendar'), { ssr: false });
-
 const CalendarPage = () => {
-  const activitiesDates = activitiesData.map((activity) => activity.date);
-  const isActivityDate = (date) => {
-    const dateString = date.toISOString().split('T')[0];
-    return activitiesDates.includes(dateString);
+  const { data: activities } = useSWR('/api/activities', fetcher);
+  console.log(activities);
+  const [value, setValue] = useState(new Date());
+  const formattedDate = value.toISOString().split('T')[0];
```

Comment:
maybe I am overlooking something it it seems like you are not using this variable anywhere in this file? 
Correct me If I am wrong!